aws-api/src/test/java/com/amplifyframework/api/aws/auth/OwnerBasedAuthTest.java

```diff
@@ -0,0 +1,351 @@
+
+    /**
+     * Sets up the test.
+     * @throws ApiException On failure to configure plugin
+     * @throws IOException On failure to start web server
+     */
+    @Before
+    public void setup() throws ApiException, IOException {
+        webServer = new MockWebServer();
+        webServer.start(8080);
+        baseUrl = webServer.url("/");
+        cognitoProvider = new FakeCognitoAuthProvider();
+        oidcProvider = new FakeOidcAuthProvider();
+        configurePlugin();
+    }
+
+    /**
+     * Stop the {@link MockWebServer} that was started in {@link #setup()}.
+     * @throws IOException On failure to shutdown the MockWebServer
+     */
+    @After
+    public void cleanup() throws IOException {
+        webServer.shutdown();
+    }
+
+    private void configurePlugin() throws ApiException {
+        ApiAuthProviders providers = ApiAuthProviders.builder()
+                .cognitoUserPoolsAuthProvider(cognitoProvider)
+                .oidcAuthProvider(oidcProvider)
+                .build();
+        JSONObject configuration = new JSONObject();
+        try {
+            configuration = configuration
+                .put(GRAPHQL_API_WITH_API_KEY, new JSONObject()
+                        .put("endpointType", "GraphQL")
+                        .put("endpoint", baseUrl.url())
+                        .put("region", "us-east-1")
+                        .put("authorizationType", "API_KEY")
+                        .put("apiKey", "FAKE-API-KEY"))
+                .put(GRAPHQL_API_WITH_COGNITO, new JSONObject()
+                        .put("endpointType", "GraphQL")
+                        .put("endpoint", baseUrl.url())
+                        .put("region", "us-east-1")
+                        .put("authorizationType", "AMAZON_COGNITO_USER_POOLS"))
+                .put(GRAPHQL_API_WITH_OIDC, new JSONObject()
+                        .put("endpointType", "GraphQL")
+                        .put("endpoint", baseUrl.url())
+                        .put("region", "us-east-1")
+                        .put("authorizationType", "OPENID_CONNECT"));
+        } catch (JSONException exception) {
+            // This shouldn't happen...
+        }
+
+        plugin = new AWSApiPlugin(providers);
+        plugin.configure(configuration, ApplicationProvider.getApplicationContext());
+    }
+
+    /**
+     * Test that owner argument fails to be appended to subscription request if
+     * the authorization mode is not OIDC compliant.
+     */
+    @Test
+    public void ownerArgumentNotAddedWithApiKey() {
+        // Set API to use API key auth mode
+        apiName = GRAPHQL_API_WITH_API_KEY;
+
+        // Attempting to subscribe to a model with owner-based auth with API key auth mode.
+        GraphQLRequest<OwnerAuth> request = ModelSubscription.onCreate(OwnerAuth.class);
+        GraphQLOperation<OwnerAuth> operation = subscribe(request);
+
+        // Subscription should fail at pre-processing
+        assertNull(operation);
```

Comment:
Is returning `null` a reasonable failure mode? Why doesn't it throw a `BadAuthException` or something to that effect?

Reply:
API plugin returns a `null` operation when encountering a failure even before sending in a network request. I can also do an assertion on the failure callback, but since the above behavior is already set in stone, I figured it would be redundant. Also the helper method `subscribe(GraphQLRequest)` was written to simplify a lot of the logic behind dealing with callbacks in this test (which mostly deals with request decoration rather than the actual result of the operation), so I would rather not add a layer of complexity here.